include/ggl/cbmc.h

```diff
@@ -0,0 +1,63 @@
+// aws-greengrass-sdk-lite - Lightweight AWS IoT Greengrass SDK
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef GGL_CBMC_H
+#define GGL_CBMC_H
+
+//! Macros for CBMC function contracts
+
+#ifndef __CPROVER__
+#define CBMC_CONTRACT(...) CBMC_CONTRACT__STRIP(__VA_ARGS__)
+#define CBMC_CONTRACT__STRIP(...)
+#else
+
+#include <ggl/macro_util.h>
+
+#define CBMC_CONTRACT(...) GGL_MACRO_FOREACH(CBMC_CONTRACT__STMT, , __VA_ARGS__)
+#define CBMC_CONTRACT__STMT(arg) CBMC_CONTRACT__##arg
+
+#define CBMC_CONTRACT__requires __CPROVER_requires
+#define CBMC_CONTRACT__ensures __CPROVER_ensures
+#define CBMC_CONTRACT__assigns __CPROVER_assigns
+#define CBMC_CONTRACT__frees __CPROVER_frees
+#define CBMC_CONTRACT__invariant __CPROVER_loop_invariant
+#define CBMC_CONTRACT__decreases __CPROVER_decreases
+
+#define cbmc_return __CPROVER_return_value
+
+#define cbmc_old(v) __CPROVER_old(v)
+#define cbmc_loop_entry(v) __CPROVER_loop_entry(v)
+
+#define cbmc_object_whole __CPROVER_object_whole
+#define cbmc_object_from __CPROVER_object_from
+#define cbmc_object_upto __CPROVER_object_upto
+
+#define cbmc_restrict(ptr, ...) \
+    __CPROVER_is_fresh(ptr, sizeof(*ptr) __VA_OPT__(*(__VA_ARGS__)))
```

Comment:
Why not name it cbmc_is_fresh?
'Restrict' was not clear when reading this for me and I had to go find this definition. Am I missing something?

Reply:
Also, what would the VA_ARGS be used for?

Reply:
Is fresh is kinda confusing since fresh isn't a C concept. It makes the pointer point to a unique allocation, so restrict fits better (any parameter that must be is_fresh could be marked restrict). The __VA_ARGS__ so you could pass it multiple pointers, all of which must be unique allocations.

Reply:
mlkem-native calls it no_alias: https://github.com/pq-code-package/mlkem-native/blob/4ca8f91876383c61b13667fd143480d483165f83/mlkem/src/cbmc.h#L58